packages/styles/scss/components/dialog/_dialog.scss

```diff
@@ -13,130 +13,171 @@
 @use '../../spacing' as *;
 @use '../../theme' as *;
 @use '../../type' as *;
-@use '../../utilities/ai-gradient' as *;
 @use '../../utilities/convert';
 @use '../../utilities/component-reset';
 @use '../../utilities/focus-outline' as *;
 @use '../../utilities/high-contrast-mode' as *;
+@use '../../utilities/update_fields_on_layer' as *;
+@use '../../utilities/z-index' as *;
 
 /// Dialog styles
 /// @access public
 /// @group dialog
 @mixin dialog {
   .#{$prefix}--dialog {
+    position: fixed;
+    z-index: z('modal');
```

Comment:
this is not needed, dialogs are automatically promoted to the [top layer](https://developer.mozilla.org/en-US/docs/Glossary/Top_layer)
```suggestion
```